lib/Runtime/Debug/DiagStackFrame.cpp

```diff
@@ -88,6 +89,306 @@ namespace Js
         return this->GetFunction()->LoadRootObject();
     }
 
+    BOOL DiagStackFrame::IsStrictMode()
+    {
+        Js::JavascriptFunction* scopeFunction = this->GetJavascriptFunction();
+        return scopeFunction->IsStrictMode();
+    }
+
+    BOOL DiagStackFrame::IsThisAvailable()
+    {
+        Js::JavascriptFunction* scopeFunction = this->GetJavascriptFunction();
+        return !scopeFunction->IsLambda() || scopeFunction->GetParseableFunctionInfo()->GetCapturesThis();
+    }
+
+    Js::Var DiagStackFrame::GetThisFromFrame(Js::IDiagObjectAddress ** ppOutAddress, Js::IDiagObjectModelWalkerBase * localsWalker)
```

Comment:
Yes see my full review request

---
In reply to: [62773327](https://github.com/Microsoft/ChakraCore/pull/926#discussion_r62773327) [](ancestors = 62773327)